src/components/CurrencyLogoBridge/index.tsx

```diff
@@ -0,0 +1,135 @@
+
+const StyledLogo = styled(Logo)<{ size: string }>`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  width: ${({ size }) => size};
+  height: ${({ size }) => size};
+  border-radius: ${({ size }) => size};
+`
+// looading is for purpose, for some reason DevTools screams that 'loading' is expected to be a string
+const Wrapper = styled.div<{ size: string; marginRight: number; marginLeft: number; looading?: boolean }>`
+  position: relative;
+  width: ${({ size }) => size};
+  height: ${({ size }) => size};
+  margin-right: ${({ marginRight }) => marginRight}px;
+  margin-left: ${({ marginLeft }) => marginLeft}px;
+  border-radius: ${({ size }) => size};
+
+  &::after {
+    content: '';
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    ${({ size }) => `width: calc(${size} - 1px)`};
+    ${({ size }) => `height: calc(${size} - 1px)`};
+    background-color: ${props => (props.looading ? 'transparent' : props.theme.white)};
+    border-radius: 50%;
+    z-index: -1;
+  }
+`
+
+const NATIVE_CURRENCY_LOGO: { [chainId in ChainId]: string } = {
+  [ChainId.ARBITRUM_ONE]: EtherLogo,
+  [ChainId.ARBITRUM_RINKEBY]: EtherLogo,
+  [ChainId.MAINNET]: EtherLogo,
+  [ChainId.RINKEBY]: EtherLogo,
+  [ChainId.XDAI]: XDAILogo
+}
+
+export default function CurrencyLogo({
+  currency,
+  size = '24px',
+  style,
+  className,
+  loading,
+  marginRight = 0,
+  marginLeft = 0,
+  chainIdOverride
+}: {
+  currency?: Currency
+  size?: string
+  style?: React.CSSProperties
+  className?: string
+  loading?: boolean
+  marginRight?: number
+  marginLeft?: number
+  chainIdOverride?: ChainId
+}) {
+  const { chainId } = useActiveWeb3React()
+  const selectedChainId = chainIdOverride || chainId
+  const nativeCurrencyLogo = NATIVE_CURRENCY_LOGO[(selectedChainId as ChainId) || ChainId.MAINNET]
+  const wrappedTokenInfo = useBridgeTokenInfo(currency, selectedChainId)
+  const uriLocations = useHttpLocations(
+    currency instanceof WrappedTokenInfo ? currency.logoURI : !!wrappedTokenInfo ? wrappedTokenInfo.logoURI : undefined
+  )
+
+  const srcs: string[] = useMemo(() => {
+    if (currency && Currency.isNative(currency) && !!nativeCurrencyLogo) return [nativeCurrencyLogo]
+    if (currency instanceof Token) {
+      if (Token.isNativeWrapper(currency)) return [nativeCurrencyLogo]
+      if (chainId && DXD[chainId] && DXD[chainId].address === currency.address) return [DXDLogo]
+      if (chainId && SWPR[chainId] && SWPR[chainId].address === currency.address) return [SWPRLogo]
+      return [getTokenLogoURL(currency.address), ...uriLocations]
+    }
+    if (currency instanceof Token)
+      return [
+        `${window.location.origin}${
+          carrotListLogoUrl.startsWith('.') ? carrotListLogoUrl.substring(1) : carrotListLogoUrl
+        }`
+      ]
+    return []
+  }, [chainId, currency, nativeCurrencyLogo, uriLocations])
```

Comment:
2 things about this block

1. I see (currency instanceof Token) twice at the same scope. Is there a good reason for that?
2. I see 2 very similar ifs for DXD and SWPR. Are these the only 2 we need to consider? If not, maybe it would be worth coming up with a function to generalise this.

Reply:
I think we should verify this with @luzzif . We thing 2nd block condition should be negated, then it would make sense.